HeterogeneousCore/CUDACore/interface/CUDAScopedContext.h

```diff
@@ -148,8 +148,8 @@ class CUDAScopedContextProduce : public impl::CUDAScopedContextGetterBase {
   explicit CUDAScopedContextProduce(const CUDAProductBase& data) : CUDAScopedContextGetterBase(data) {}
 
   /// Constructor to re-use the CUDA stream of acquire() (ExternalWork module)
-  explicit CUDAScopedContextProduce(CUDAContextState& token)
-      : CUDAScopedContextGetterBase(token.device(), std::move(token.streamPtr())) {}
+  explicit CUDAScopedContextProduce(CUDAContextState& state)
+      : CUDAScopedContextGetterBase(state.device(), state.releaseStreamPtr()) {}
```

Comment:
why is the move not needed any more ?
or, why was it needed in the first place ?

Reply:
Because I decided it to be less confusing to have the `move` inside `CUDAContextState::releaseStreamPtr()` than have `CUDAContextState` to give non-`const`-reference to its internal `shared_ptr` and do the `move` here.

See also https://github.com/cms-patatrack/cmssw/pull/404/files#r343877085